### DISCUSSION TITLE
Where should I put the certificates files for integration with third party service?

### DISCUSSION THREAD

**Discussion by siarheipashkevich** [laravel/framework#50899] [Q&A]
Hi, all.

I'm working on the issuing Apple Wallet passes, and for integration with Apple Wallet I need to use certificates. My question is - where the best place to put these certificates for avoiding security issues?

Currently I have S3 bucket, but I'm not sure if this is the best solution for storing certificates.

Thanks for advices!

**Comment by ghost** [comment]
Aws secrets?

**Reply by siarheipashkevich** [reply]
No, Apple certificates

**Reply by ghost** [reply]
I meant in aws secrets would not be ok?

**Reply by siarheipashkevich** [reply]
Ah, sorry. I'm not using AWS, I'm using DigitalOcean with Spaces (S3). Do you think is it ok to save credentials on buckets?

**Reply by ghost** [reply]
No. If you are using gitlab you can store them in gitlab variables and from there, on deploy you could put them in the environment of the docker container, not in .env file. It is just an idea. Maybe not the best one.
